lib/src/rules/conformance/banned_property_write.dart

```diff
@@ -0,0 +1,240 @@
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+import '../../analyzer.dart';
+import 'conformance_rule.dart';
+import 'interop_helpers.dart';
+import 'web_bindings.dart';
+
+class _BannedPropertyWriteCode extends SecurityLintCode {
+  _BannedPropertyWriteCode(
+      {required String name,
+      required String problemMessage,
+      required String correctionMessage})
+      : super(name, problemMessage, correctionMessage: correctionMessage);
+}
+
+class BannedPropertyWrite extends ConformanceRule {
+  final String? nativeType;
+  final String? nativeProperty;
+  final String? dartHtmlType;
+  final String? dartHtmlProperty;
+  BannedPropertyWrite(
+      {required String name,
+      required String description,
+      required String details,
+      required this.nativeType,
+      required this.nativeProperty})
+      : dartHtmlType = null,
+        dartHtmlProperty = null,
+        super(name: name, description: description, details: details);
+
+  // This constructor is meant specifically to disallow specific `dart:html`
+  // APIs that aren't `external`. Using this without a separate rule disallowing
+  // the native property you are interested in would be insufficient, as this
+  // does not cover the interop case.
+  BannedPropertyWrite.withDartHtmlTypes(
+      {required String name,
+      required String description,
+      required String details,
+      required this.dartHtmlType,
+      required this.dartHtmlProperty})
+      : nativeType = null,
+        nativeProperty = null,
+        super(name: name, description: description, details: details);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _BannedPropertyWriteVisitor(this);
+    registry.addCompilationUnit(this, visitor);
+    registry.addAssignmentExpression(this, visitor);
+    registry.addMethodInvocation(this, visitor);
+  }
+}
+
+class _BannedPropertyWriteVisitor extends SimpleAstVisitor<void> {
+  final BannedPropertyWrite rule;
+
+  _BannedPropertyWriteVisitor(this.rule);
+
+  @override
+  void visitCompilationUnit(CompilationUnit node) {
+    computeHtmlBindings(node.declaredElement?.library);
+  }
+
+  @override
+  void visitAssignmentExpression(AssignmentExpression expression) {
+    var leftHandSide = expression.leftHandSide;
+    DartType? dartTargetType;
+    late String dartProperty;
+    if (leftHandSide is PropertyAccess) {
+      dartTargetType = leftHandSide.realTarget.staticType;
+      dartProperty = leftHandSide.propertyName.name;
+    } else if (leftHandSide is PrefixedIdentifier) {
+      dartTargetType = leftHandSide.prefix.staticType;
+      dartProperty = leftHandSide.identifier.name;
+    } else {
+      return;
+    }
+
+    var dartTargetTypeElement = dartTargetType?.element;
```

Comment:
`dartTargetTypeElement` is used in line 89. I could check that `dartTargetType` is `null` and then `return` and then do this assignment, but I think that'll just be adding an extra line since `element` is still nullable and needs to be checked.

Reply:
Yep. I'd missed the use on line 89.